modules/rtbhouseBidAdapter.js

```diff
@@ -96,30 +63,257 @@ export const spec = {
     };
   },
   interpretResponse: function (serverResponse, originalRequest) {
-    serverResponse = serverResponse.body;
-    const bids = [];
-
-    if (utils.isArray(serverResponse)) {
-      serverResponse.forEach(serverBid => {
-        if (serverBid.price !== 0) {
-          const bid = {
-            requestId: serverBid.impid,
-            mediaType: BANNER,
-            cpm: serverBid.price,
-            creativeId: serverBid.adid,
-            ad: serverBid.adm,
-            width: serverBid.w,
-            height: serverBid.h,
-            ttl: 55,
-            netRevenue: true,
-            currency: 'USD'
-          };
-          bids.push(bid);
-        }
-      });
+    const responseBody = serverResponse.body;
+    if (!utils.isArray(responseBody)) {
+      return [];
     }
+
+    const bids = [];
+    responseBody.forEach(serverBid => {
+      if (serverBid.price === 0) {
+        return;
+      }
+      if (serverBid.adm.startsWith('{"native')) {
+        bids.push(interpretNativeBid(serverBid));
+      } else {
+        bids.push(interpretBannerBid(serverBid));
```

Comment:
Shouldn't you test here for `serverBid.adm.startsWith('{"banner')` and throw error otherwise?

Reply:
No, because banner response most often starts with "<iframe" and before we handle only banners so it makes sense that default will be banner, no risk of breaking something.

Reply:
+1 - I'd prefer to go pythonic way and try-parse the supposed JSON, catch exception and go from there
or even better - find the source bid's type & match based on the source bid's type - ie bid is Native -> expect Native; If is not Native - raise exception & do something from there

Reply:
I know it's ugly but can't find better way, try...catch is hard to do properly because JSON.parse() throws SyntaxError. If you know how to do it better I'll be glad.